### PR TITLE
Improve title consistency

### LINKED ISSUE #2242: Make title entries consistent
We should do a pass over the `title:` part of the YAML at the top of every page. I noticed two small things that are wrong:

1. Pages with colons in the names sometimes capitalize the word that follows the colon, but sometimes don't. Examples: **Effective Dart: Usage** vs. **Overview: the Dart language**
2. Pages without colons in the names sometimes use unneeded quotation marks. Example: `title: "Remove DOM elements"`

My natural inclination is to capitalize after the colon, but I think that might be against Google style. Someone should check.

Here's how to find all the titles with quotation marks (i.e. both 1 & 2, above), if you're in the top directory of the site-www repo:

```
fgrep -R "title: \"" src
```

### PR DESCRIPTION
**Pull request by parlough**
Follows the standards for titles and capitalization from the [Google style guide](https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings).

Fixes #2242

### CODE REVIEW

File: src/_data/side-nav.yml [hunk 1]
```diff
@@ -165,7 +165,7 @@
           permalink: /tutorials/web/fetch-data
         - title: Low-level web programming
           children:
-          - title: Connect Dart & HTML
+          - title: Connect Dart and HTML
```
Comment (kwalrath) [inline src/_data/side-nav.yml:168]: We intentionally use & in the sidenav and tabs to save space. Might as well be consistent.

```suggestion
          - title: Connect Dart & HTML
```

File: src/_tutorials/server/cmdline.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ nextpage:
   title: Write HTTP clients & servers
```
Comment (kwalrath) [inline src/_tutorials/server/cmdline.md:6]:
```suggestion
  title: Write HTTP clients and servers
```

File: src/map.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-title: "Sitemap"
-description: "Human Readable Sitemap"
+title: Sitemap
+description: Human Readable Sitemap
```
Comment (kwalrath) [inline src/map.html:3]:
```suggestion
description: Human-readable sitemap
```

File: src/server/libraries.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Command-line & server libraries and packages"
+title: Command-line & server libraries and packages
```
Comment (kwalrath) [inline src/server/libraries.md:2]:
```suggestion
title: Command-line and server libraries and packages
```

File: src/_data/side-nav.yml [hunk 5]
```diff
@@ -260,7 +260,7 @@
       permalink: /resources
     - title: Books
       permalink: /resources/books
-    - title: DartPad in tutorials&#58; best practices
+    - title: DartPad in tutorials&#58; Best practices
```
Comment (kwalrath) [inline src/_data/side-nav.yml:263]:
```suggestion
    - title: "DartPad in tutorials: Best practices"
```

Comment (kwalrath) [reply src/_data/side-nav.yml:263]: I didn't know what `&#58;` was!

**Review by kwalrath — COMMENTED** [review]
Finally I got to this. Good stuff, @parlough!

I had some quibbles, and I think we might've missed some ampersands, but I like these rules (which are pretty close to the ones you used):

* Use "and" (not "&") in page titles, but you can change that to "&" in places where space is constrained (sidenav, tab titles aka short-title, maybe page set navigation).
* In a page title that has a colon, capitalize the word that follows the colon.
* Avoid unnecessary quotation marks in the YAML header.

(Did I miss any?)

**Comment by parlough** [comment]
@kwalrath Thanks for the suggestions, I modified one more to match one of the other changes you suggested.

I can follow up by adding some title guidelines to https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#top-matter-yaml

**Review by kwalrath — APPROVED** [review]
LGTM!

**Comment by kwalrath** [comment]
> I can follow up by adding some title guidelines to https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#top-matter-yaml

Sounds perfect. Thank you!

**Pull request merged**